sim/battle.ts

```diff
@@ -2228,11 +2228,11 @@ export class Battle {
 		if (this.activePerHalf > 2) {
 			if (move.target === 'adjacentFoe' || move.target === 'normal' || move.target === 'randomNormal') {
 				// even if a move can target an ally, auto-resolution will never make it target an ally
-				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your all
+				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your ally
 				const adjacentFoes = pokemon.adjacentFoes();
 				if (adjacentFoes.length) return this.sample(adjacentFoes);
-				// no valid target at all, return a possibly-fainted foe for any possible redirection
-				return pokemon.side.foe.active[0];
+				// no valid target at all, return slot directly across for any possible redirection
```

Comment:
```js
			// copies across in doubles/triples
			// (also copies across in multibattle and diagonally in free-for-all,
			// but side.foe already takes care of those)
const target = pokemon.side.foe.active[pokemon.side.foe.active.length - 1 - pokemon.position];
```

https://github.com/smogon/pokemon-showdown/blob/4291ccb51bcee3f33c4b18b136f92dc439b65143/data/abilities.ts

The comment is inaccurate, but the implementation looks good.